python_modules/dagster/dagster/_config/structured_config.py

```diff
@@ -219,8 +226,18 @@ def create_io_manager_to_pass_to_user_code(self, context) -> IOManager:
 
 PydanticShapeType: TypeAlias = int
 
+MAPPING_TYPES = {SHAPE_MAPPING, SHAPE_DICT}
+MAPPING_KEY_TYPE_TO_SCALAR = {
+    StringSource: str,
+    IntSource: int,
+    BoolSource: bool,
+    ConfigFloatInstance: float,
+}
```

Comment:
Oh now I understand. We were mapping the mapping types here. That's what got me a bit confused.